### PR TITLE
Add score averaging and convergence early stop

### PR DESCRIPTION
**Pull request by CedricHwong**
## What does this PR do?

Overview:
This PR extends GradNAS gradient scoring with 
(1) optional score averaging 
(2) early stopping when gradient scores converge. New search config keys are added with defaults: average_scores, score_convergence_tol, score_convergence_patience, and score_convergence_min_updates. The gradient estimation loop now exits early once all hparam score trackers converge. Unit tests cover the averaging math, convergence boundaries, and disabled-convergence behavior.

## Usage

```python
  import modelopt.torch.prune as mtp

  # Example: enable/adjust score averaging + convergence early stop
  pruned_model, search_history = mtp.prune(
      model=model,
      mode="gradnas",
      constraints={"flops": "90%"},
      dummy_input=dummy_input,
      config={
          "data_loader": train_loader,
          "loss_func": loss_func,
          # New options:
          "average_scores": True,
          "score_convergence_tol": 1e-3,
          "score_convergence_patience": 5,
          "score_convergence_min_updates": 10,
      },
  )
```

## Testing

  - pytest tests/unit/torch/prune/test_gradnas.py
  - pre-commit run --all-files

  ## Before your PR is "Ready for review"
  

  - Make sure you read and follow Contributor guidelines (https://github.com/NVIDIA/Model-
    Optimizer/blob/main/CONTRIBUTING.md) and your commits are signed.
  - Is this change backward compatible?: No Default GradNAS behavior now averages gradient scores and may stop early on convergence, which can change score scale and number of batches processed vs. previous behavior.
  - Did you write any new necessary tests?: Yes
  - Did you add or update any necessary documentation?: No
  - Did you update Changelog (https://github.com/NVIDIA/Model-Optimizer/blob/main/CHANGELOG.rst)?: No

  ## Additional Information

  N/A

### CODE REVIEW

**Comment by kevalmorabia97** [comment]
Hi @CedricHwong we dont use GradNAS anymore these days. Can you please share your pruning use case and what improvements you see after this change?

**Comment by realAsma** [comment]
https://github.com/NVIDIA/Model-Optimizer/pull/734#issuecomment-3711304155

same as  @kevalmorabia97 's comments. Could you please elaborate your use case and what improvements you observe?

**Comment by kevalmorabia97** [comment]
Closing as no further response
